tests/command_line/test_chargemol_caller.py

```diff
@@ -86,31 +80,61 @@ def mock_xyz(tmp_path: Path):
 
 
 def test_parse_chargemol(monkeypatch, mock_xyz):
-    monkeypatch.setattr("pymatgen.command_line.chargemol_caller.download", fake_download)
+    monkeypatch.setattr(
+        "pymatgen.command_line.chargemol_caller._download_and_unzip_atomic_densities",
+        fake_download,
+    )
 
     ca = ChargemolAnalysis(path=str(mock_xyz), run_chargemol=False)
     ca._atomic_densities_path = os.path.expanduser("~/.cache/pymatgen/ddec/fake_file")
 
     assert ca.ddec_charges == [0.8432, -0.8432]
 
 
-def test_auto_download(monkeypatch):
-    # mock getenv to simulate that DDEC6_ATOMIC_DENSITIES_DIR is not set
-    monkeypatch.setattr(os, "getenv", lambda *args, **kwargs: None)
+def fake_download(self, version: str = "latest", verbose: bool = True) -> None:
+    extraction_path = "~/.cache/pymatgen/ddec"
```

Comment:
Be careful with path handling. You don't want to impose a given OS.

I recommend using the `pathlib` library instead.

Reply:
Actually, forward slashes work under windows. I've only encountered a few arcane cases where they cause problems. See the `pmg` test suite. Runs on Windows in GHA and it's all forward slashes.

Reply:
The `~` is not valid on Windows, although I guess you get out of that pickle by doing `os.path.expanduser`.

Reply:
Doing `os.path.expanduser(extraction_path)` expands to `C:\\Users\\asros/.cache/pymatgen/ddec'` which will work with `os.makedir` since `os` is smart. So, tests will pass, but you're playing with fire a bit... 😅

Reply:
Yeah, i'm running a live experiment. So far it's only caused issues 2 or 3 times and they were niche. Feel like so far the readability gain from not using `os.path.join` was well worth it. Need to reconsider if sth non-niche comes up.

Reply:
Feel free to ignore these comments then @chiang-yuan. Good to know, @janosh.

Reply:
Thanks @Andrew-S-Rosen . I actually prefer `pathlib` since it is a newer standard over `os.path`. I was just following what we already have in the code 😉 Will change it whenever appropriate :)

Reply:
Seems it doesn't really matter :)

Reply:
while `pathlib` has some nice features, i find it a bit too OOP with resulting perf overhead compared to `os.path` (see e.g. https://youtu.be/tFrh9hKMS6Y). although admittedly the speed diff doesn't matter in practice.